src/traces/waterfall/attributes.js

```diff
@@ -0,0 +1,68 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+var barAttrs = require('../bar/attributes');
+
+module.exports = {
+
+    x: barAttrs.x,
+    x0: barAttrs.x0,
+    dx: barAttrs.dx,
+    y: barAttrs.y,
+    y0: barAttrs.y0,
+    dy: barAttrs.dy,
+
+    r: barAttrs.r,
+    t: barAttrs.t,
+
+    text: barAttrs.text,
```

Comment:
From https://github.com/plotly/plotly.js/issues/3037#issuecomment-459806749, 

> also note we will want to support `text` display which should be the delta value, not the absolute value (and the start, delta and end values should be available in the hovertemplate if possible).

Is this the case?

Moreover, we should probably have `increasing.*textfont` and `decreasing.*textfont` container. I can imagine users wanting different text font colours for increasing vs decreasing bars.

Reply:
Maybe we could implement https://github.com/plotly/plotly.js/issues/85 if the to-be-added logic is sufficient similar in the stacked bar case.